### PR TITLE
CIRC-8734: Fix issue with datasource backend retrying requests

### PR DESCRIPTION
**Pull request by dhaifley**
* Fixed a bug where the API client used by the datasource could retry on response errors multiple times, potentially causing a large number of API requests to be generated.

### CODE REVIEW

File: pkg/irondb-backend.go [hunk 1]
```diff
@@ -78,9 +78,10 @@ func (td *SampleDatasource) QueryData(ctx context.Context, req *backend.QueryDat
 		queryURL := req.PluginContext.DataSourceInstanceSettings.URL
 
 		td.circ, err = circ.New(&circ.Config{
-			TokenKey: key,
-			TokenApp: "circonus-irondb-datasource",
-			URL:      queryURL,
+			TokenKey:   key,
+			TokenApp:   "circonus-irondb-datasource",
+			URL:        queryURL,
+			MaxRetries: 0, // Do not retry 5XX error as doing so could lead to a flood of API requests.
```
Comment (bdunavant) [inline pkg/irondb-backend.go:84]:
```suggestion
			MaxRetries: 0, // Do not retry 5XX error as Grafana is responsible for retry behavior
```

Comment (dhaifley) [reply pkg/irondb-backend.go]: I have updated the comment to make it better grammatically, and to replace the language you suggested.

File: dist/module.js [hunk 2]
```diff
@@ -1,3 +1,3 @@
...
...on(e){r||(r=!0,i(this,t,n(e)),r=!1)}))),t}},function(e,t,r){"use strict";var n=r(7),a=r(13).Symbol;e.exports=function(e){return Object.defineProperties(e,{hasInstance:n("",a&&a.hasInstance||e("hasInstance")),isConcatSpreadable:n("",a&&a.isConcatSpreadable||e("isConcatSpreadable")),iterator:n("",a&&a.iterator||e("iterator")),match:n("",a&&a.match||e("match")),replace:n("",a&&a.replace||e("replace")),search:n("",a&&a.search||e("search")),species:n("",a&&a.species||e("species")),split:n("",a&&a.split||e("split")),toPrimitive:n("",a&&a.toPrimitive||e("toPrimitive")),toStringTag:n("",a&&a.toStringTag||e("toStringTag")),unscopables:n("",a&&a.unscopables||e("unscopables"))})}},function(e,t,r){"use strict";var n=r(7),a=r(30),i=Object.create(null);e.exports=function(e){return Object.defineProperties(e,{for:n((function(t){return i[t]?i[t]:i[t]=e(String(t))})),keyFor:n((function(e){var t;for(t in a(e),i)if(i[t]===e)return t}))})}},function(e,t,r){"use strict";var n=Object.prototype.toString,a=n.call(function(){return arguments}());e.exports=function(e){return n.call(e)===a}},function(e,t,r){"use strict";var n=Object.prototype.toString,a=RegExp.prototype.test.bind(/^[object [A-Za-z0-9]*Functio
```
[the remark this answers is not in the document]
Comment (dhaifley) [reply dist/module.js]: Yeah, I ran a yarn build/test when I made this and it updated them, but they do not need to be in this PR. I have removed those changes. Thanks!

**Review by graefen — APPROVED** [review]
LGTM

**Pull request merged**